driver_4_lights.ino

```diff
@@ -1,225 +1,345 @@
-/*
-  Blink
-  Turns on an LED on for one second, then off for one second, repeatedly.
-
-  Most Arduinos have an on-board LED you can control. On the UNO, MEGA and ZERO
-  it is attached to digital pin 13, on MKR1000 on pin 6. LED_BUILTIN is set to
-  the correct LED pin independent of which board is used.
-  If you want to know what pin the on-board LED is connected to on your Arduino model, check
-  the Technical Specs of your board  at https://www.arduino.cc/en/Main/Products
+//Light machine logic
 
-  This example code is in the public domain.
+/*
+  State button (LED to Human)       meaning
+  0 Nothing (LED 1/sec blink)       Just looping looking for input
+  1 StepLights (LED 3/sec blink)    Can step through lights
+  2 ManualStepThrough (LED 10/sec)  When you push camera button, light on and picture
+  3 AutoStepThrough (LED on)        Goes through all lights and all pictures
 
-  modified 8 May 2014
-  by Scott Fitzgerald
+  If you are in the middle of a state, pressing state returns you to state 0 (Nothing)
 
-  modified 2 Sep 2016
-  by Arturo Guadalupi
+  Camera button: Push to take picture while in state
+  1 (StepLights) changes the lights
+  2 (ManualStepThrough) change the lights and camera
+  3 (AutoStepThrough) collect pictures with light automatically for all lights
 
-  modified 8 Sep 2016
-  by Colby Newman
 */
 
-
-// constants won't change. They're used here to
-// set pin numbers:
-const int pinButtonState = 0;    // the number of the pushbutton pin
-const int pinButtonCamera = 8;   //button for the camera
-const int ledPin = 13;      // the number of the LED pin
+const int pinButtonState = 8; // the number of the pushbutton pin
+const int pinButtonCamera = 10; //button for the camera
+const int ledPin = 13; // the number of the LED pin
+const int indicatorLed = 12; //the led that shows what state you are in on circuit board
+const int cameraShutter = 11;
+const int ledTR = 3;   //TR light on circut board
+const int ledTL = 4;   //TL light on cirut board
+const int ledBR = 5;   //BR light on circut board
+const int ledBL = 6;   //BL light on circut board
+const int ledAll = 10; // all the lights are ON (not on circut board)
+
+const int whichLightOn; //which of the 4 lights are on
+
+const int shutterLag = 50;
```

Comment:
A common convention for time variables is to append the resolution e.g. ms for milliseconds.

This might read like: 
const int SHUTTER_LAG_MS = 50;  // 50 ms between when we send the shutter signal and when the frame is taken.